File: BazaMobileHW/src/main/NotesApp/src/Storage.kt
class Storage<T : IdentifiableModel> {
    private val items = mutableListOf<T>()

    fun add(item: T) {
        items.add(item)
    }

    fun removeById(id: String) {
        items.removeAll { it.id == id }
    }

    fun getById(id: String): T? {
        return items.firstOrNull { it.id == id }
    }

    fun update(item: T) {
        val index = items.indexOfFirst { it.id == item.id }
        if (index != -1) {
            items[index] = item
        }
    }

    fun getAll(): List<T> {
        return items.toList()
    }
}

File: NotesApp/Storage.swift
import Foundation
protocol IdentifableModel {
    var id : String { get}
}

class Storage<T:IdentifableModel>{
     var Items: [T] = []

    
    func add(_ item : T){
        Items.append(item)
    }
    
    func remove(ById id : String){
        Items.removeAll{$0.id == id}
    }
    
    func value (by id:String) -> T? {
        return Items.first(where: {$0.id == id})
    }
    
    func update (_ item: T){
        for i in Items.indices {
            if Items[i].id == item.id{
                Items[i] = item
            }
        }
    }
    
    func getAll() -> [T] {
            return Items
        }
    
    
    
}

